crates/handler/src/service_route.rs

```diff
@@ -54,7 +54,7 @@ impl DerefMut for ServiceRouteTable {
 
 impl ServiceRouteTable {
     /// Call the GraphQL query of the specified service.
-    #[instrument(err(Debug), ret, level = "trace")]
+    #[instrument(err(Debug), skip(header_map), ret, level = "trace")]
```

Comment:
Just one thing: not sure about leaving `request: Request,` in `pub async fn query` trace as it could potentially undisclose secret/sensitive information- ids, payment info...